pkg/controllers/workgenerator/controller.go

```diff
@@ -220,6 +221,33 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 	return controllerruntime.Result{}, syncErr
 }
 
+// updateBindingStatusWIthRetry sends the update request to API server with retry.
+func (r *Reconciler) updateBindingStatusWithRetry(ctx context.Context, resourceBinding *fleetv1beta1.ClusterResourceBinding) error {
+	// Retry only for specific errors or conditions
+	errAfterRetries := retry.RetryOnConflict(retry.DefaultRetry, func() error {
+		var latestBinding fleetv1beta1.ClusterResourceBinding
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(resourceBinding), &latestBinding); err != nil {
+			return err
+		}
+		// Work generator is the only controller that updates conditions excluding rollout started which is updated by rollout controller.
+		if rolloutCond := latestBinding.GetCondition(string(fleetv1beta1.ResourceBindingRolloutStarted)); rolloutCond != nil {
+			resourceBinding.SetConditions(*rolloutCond)
```

Comment:
The setCondition call will change the lastTransitionTime if the status has changed which is not what we want. ie. if it was bound but now its unschedued, this will mark the unscheudeuled lastTransitionTime = Now() instead of when the rollout unselect it.  This has to be an exact copy.